client/consul.go

```diff
@@ -0,0 +1,170 @@
+	IsEnterprise(ctx context.Context) (bool, error)
+}
+
+// ConsulClient is a client to the Consul API
+type ConsulClient struct {
+	*consulapi.Client
+	retry  retry.Retry
+	logger logging.Logger
+}
+
+// ConsulAgentConfig represents the response body from Consul /v1/agent/self API endpoint.
+// The response contains configuration and member information of the requested agent.
+// Care must always be taken to do type checks when casting, as structure could
+// potentially change over time.
+type ConsulAgentConfig = map[string]map[string]interface{}
+
+// NewConsulClient constructs a consul api client
+func NewConsulClient(conf *config.Config, maxRetry int) (*ConsulClient, error) {
+	consulConf := conf.Consul
+	transport := hcat.TransportInput{
+		SSLEnabled: *consulConf.TLS.Enabled,
+		SSLVerify:  *consulConf.TLS.Verify,
+		SSLCert:    *consulConf.TLS.Cert,
+		SSLKey:     *consulConf.TLS.Key,
+		SSLCACert:  *consulConf.TLS.CACert,
+		SSLCAPath:  *consulConf.TLS.CAPath,
+		ServerName: *consulConf.TLS.ServerName,
+
+		DialKeepAlive:       *consulConf.Transport.DialKeepAlive,
+		DialTimeout:         *consulConf.Transport.DialTimeout,
+		DisableKeepAlives:   *consulConf.Transport.DisableKeepAlives,
+		IdleConnTimeout:     *consulConf.Transport.IdleConnTimeout,
+		MaxIdleConns:        *consulConf.Transport.MaxIdleConns,
+		MaxIdleConnsPerHost: *consulConf.Transport.MaxIdleConnsPerHost,
+		TLSHandshakeTimeout: *consulConf.Transport.TLSHandshakeTimeout,
+	}
+
+	consul := hcat.ConsulInput{
+		Address:      *consulConf.Address,
+		Token:        *consulConf.Token,
+		AuthEnabled:  *consulConf.Auth.Enabled,
+		AuthUsername: *consulConf.Auth.Username,
+		AuthPassword: *consulConf.Auth.Password,
+		Transport:    transport,
+	}
+
+	clients := hcat.NewClientSet()
+	if err := clients.AddConsul(consul); err != nil {
+		return nil, err
+	}
+
+	logger := logging.Global().Named(loggingSystemName).Named(consulSubsystemName)
+
+	r := retry.NewRetry(maxRetry, time.Now().UnixNano())
+	c := &ConsulClient{
+		Client: clients.Consul(),
+		retry:  r,
+		logger: logger}
+	return c, nil
+}
+
+// GetLicense queries Consul for a signed license, and returns it if available
+func (c *ConsulClient) GetLicense(ctx context.Context, q *consulapi.QueryOptions) (string, error) {
+	c.logger.Debug("getting license")
+
+	desc := "consul client get license"
+	var err error
+	var license string
+
+	f := func(context.Context) error {
+		license, err = c.Operator().LicenseGetSigned(q)
+		if err != nil {
+			license = ""
+			return err
+		}
+		return nil
+	}
+
+	err = c.retry.Do(ctx, f, desc)
+
+	return license, err
+}
+
+// IsEnterprise queries Consul for information about itself, it then
+// parses this information to determine whether the Consul being
+// queried is Enterprise or OSS. Returns true if Consul is Enterprise.
+func (c *ConsulClient) IsEnterprise(ctx context.Context) (bool, error) {
+	c.logger.Debug("getting sku")
```

Comment:
There's still a couple references to "sku" here and in a couple other places. Would you mind doing a find and replace?